far/viewer.cpp

```diff
@@ -296,13 +296,29 @@ wchar_t Viewer::ZeroChar() const
 
 void Viewer::ChangeHexModeBytesPerLine(int const Amount)
 {
-	const auto NewValue =
-		std::clamp(m_HexModeBytesPerLine + Amount, s_HexModeBytesPerStripe, hex_mode_bytes_from_line_width(ScrollbarAdjustedWidth, get_char_size(m_Codepage)));
+	if (Amount == 0) return;
 
-	if (NewValue == m_HexModeBytesPerLine)
-		return;
+	// Round up to the nearest multiple of AbsAmount.
+	const auto RoundUp = [](const auto BytesPerLine, const auto AbsAmount)
+		{
+			return (BytesPerLine / AbsAmount + 1) * AbsAmount;
+		};
+
+	const int RoundedBytesPerLine = [&]()
+		{
+			if (Amount > 0) return RoundUp(m_HexModeBytesPerLine, Amount);
+
+			const auto Pivot{ RoundUp(m_HexModeBytesPerLine, -Amount) };
+			// Do the math in the inverted space X' = Pivot - X, then return to the original space.
+			return Pivot - RoundUp(Pivot - m_HexModeBytesPerLine, -Amount);
+		}();
 
-	m_HexModeBytesPerLine = NewValue;
+	const auto NewBytesPerLine =
+		std::clamp(RoundedBytesPerLine, s_HexModeBytesPerStripe, hex_mode_bytes_from_line_width(ScrollbarAdjustedWidth, get_char_size(m_Codepage)));
```

Comment:
> This PR changes the behavior of `Ctrl+Alt+Arrow` keys in Hex Mode to snap the number of bytes per line to the nearest multiple of 16 in the corresponding direction.

Does it?
Ctrl+Alt+Left eventually goes from 16 to 8
Ctrl+Alt+Right eventually goes from 16*N to whatever fits into the window.

This behaviour is inherited from the previous implementation of course and comes from this clamp, but I'm not sure it makes sense anymore: Ctrl+Alt+Keys sometimes snapping and sometimes doing something else is confusing.
Maybe just do nothing if we can't make it a multiple of 16?

Also, maybe reflect the snapping in help?
For some reason it's in at least two places:
https://github.com/FarGroup/FarManager/blob/8d1fc4d68092498130f02772df328a1aca279326/far/FarEng.hlf.m4#L2859-L2862
https://github.com/FarGroup/FarManager/blob/8d1fc4d68092498130f02772df328a1aca279326/far/FarEng.hlf.m4#L3080-L3085

Reply:
I thought about it as well. Decided to keep the old behavior, but happy to change it.

Help: ugh! I updated both `@Viewer` and `@ViewerMode` articles. Please read carefully.